lib/HeroBonus.h

```diff
@@ -316,7 +315,7 @@ class DLL_LINKAGE CAddInfo : public std::vector<si32>
 	BONUS_NAME(CATAPULT_EXTRA_SHOTS) /*val - number of additional shots, requires CATAPULT bonus to work*/\
 	BONUS_NAME(RANGED_RETALIATION) /*allows shooters to perform ranged retaliation*/\
 	BONUS_NAME(BLOCKS_RANGED_RETALIATION) /*disallows ranged retaliation for shooter unit, BLOCKS_RETALIATION bonus is for melee retaliation only*/\
-  	BONUS_NAME(SECONDARY_SKILL_VAL2) /*for secondary skills that have multiple effects, like eagle eye (max level and chance)*/  \
+	BONUS_NAME(SECONDARY_SKILL_VAL2) /*deprecated. has no effect, will be converted to actual bonus*/  \

```

Comment:
Since you're breaking save format anyway, can you move deprecated bonuses to the very end of list?

Reply:
@IvanSavenko will do. I can remove it altogether, but then many mods can crash. So, I think better to leave this as bonuses, but remove on next pass.

Reply:
Yup. I would also add some warning on loading, e.g. 
"bonus X deprecated, please replace with Y"

Reply:
In which place I should add such a warning, to affect all bonus loadings?

Reply:
@IvanSavenko done the following:
1. Removed deprecated bonuses
2. Wrote deprecated bonuses converter, which converts it to normal bonuses anyway (except some bonuses).
3. Converter will write a new str to copy-paste inside mod.